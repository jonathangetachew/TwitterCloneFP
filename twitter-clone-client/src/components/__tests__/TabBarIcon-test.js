import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import TabBarIcon from '../TabBarIcon';

jest.mock('@expo/vector-icons', () => ({
    Ionicons: 'Ionicons'
}));

describe('TabBarIcon', () => {
    jest.useFakeTimers();

    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it(`renders the icon focused`, () => {
        const tree = renderer.create(<TabBarIcon focused />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(`renders the icon normal`, () => {
        const tree = renderer.create(<TabBarIcon />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
