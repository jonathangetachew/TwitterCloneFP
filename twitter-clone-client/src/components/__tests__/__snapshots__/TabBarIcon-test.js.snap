// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon renders the icon focused 1`] = `
<Ionicons
  color="#2f95dc"
  size={26}
  style={
    Object {
      "marginBottom": -3,
    }
  }
/>
`;

exports[`TabBarIcon renders the icon normal 1`] = `
<Ionicons
  color="#ccc"
  size={26}
  style={
    Object {
      "marginBottom": -3,
    }
  }
/>
`;
